### PR TITLE
Add Node.js testing infrastructure

### PR DESCRIPTION
**Pull request by rod-glover**
This PR adds Node.js testing infrastructure. In particular:

* `package.json` and `package-lock.json` for installing Node.js test dependencies (as dev dependencies; we are _not_ turning the JS app into a Node project). In particular, we install the Node package Jest and some related packages to support writing tests of JS code.
* New file `globals-helpers.js`, plus tests, which enables test code to set up globals in the same way they are set up in production code.
* New file `condExports.js`, plus tests, which enables all files to export conditionally to the global object or as a Node module, depending on environment (production or test, i.e., Node). This supports testing apps with minimal disturbance to existing production code.
* New file `classes.js`, plus tests, which makes it easy/ier to define class-like structures in ES5. This will be used in new production code to be added in a later PR.

Note: The base branch for this PR is i110-main; see #116.

### CODE REVIEW

File: pdp/static/js/__test__/classes.test.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        });
+
+        it('speaks (prototype)', function() {
+            expect(b.speak()).toBe('I am Groot!!!');
+        });
+    });
+});
+
+
+////////////
```
Comment (rod-glover) [inline pdp/static/js/__test__/classes.test.js]: Everything from here on is some experimentation I was doing. Reviewers can ignore.

File: pdp/static/js/__test__/condExport.test.exporter.obj.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// A test user of es6export
```
Comment (rod-glover) [inline pdp/static/js/__test__/condExport.test.exporter.obj.js]: Rename: condExport

**Comment by corviday** [comment]
I was able to build the branch and run the tests just by following standard Node procedure, but I think it would be nice to add instructions on how to build the javascript half of the PDP and run the tests to the README anyway.

NPM had a couple complains that I think could be trivially fixed by adding a line to package.json, so we might as well:
```
npm WARN pdp@1.0.0 No repository field.
npm WARN pdp@1.0.0 No license field.
```

File: pdp/static/js/__test__/classes.test.js (hunk 1 above)
Comment (corviday) [reply pdp/static/js/__test__/classes.test.js]: Are you planning to remove the experimentation?

**Comment by rod-glover** [comment]
@corviday , LGTY?

**Comment by rod-glover** [comment]
Well dang, looks like I broke something.

**Comment by jameshiebert** [comment]
Looks like it installed GDAL 1.10, even though we've asked for 2.2...

**Comment by rod-glover** [comment]
OK, I didn't break it. It broke itself.

**Comment by jameshiebert** [comment]
Looks to me like something changed on the PPA and we're not getting libgdal-dev 2.2 from it anymore. Here's the last master build:

```
Get:47 http://ppa.launchpad.net/ubuntugis/ubuntugis-unstable/ubuntu trusty/main amd64 libspatialite5 amd64 4.1.1-5ubuntu2 [853 kB]
Get:48 http://ppa.launchpad.net/ubuntugis/ubuntugis-unstable/ubuntu trusty/main amd64 libopenjp2-7 amd64 2.1.0-2.1otb1~trusty1 [102 kB]
Get:49 http://ppa.launchpad.net/ubuntugis/ubuntugis-unstable/ubuntu trusty/main amd64 gdal-data all 2.2.2+dfsg-1~trusty1 [568 kB]
Get:50 http://ppa.launchpad.net/ubuntugis/ubuntugis-unstable/ubuntu trusty/main amd64 libgdal20 amd64 2.2.2+dfsg-1~trusty1 [5,110 kB]
Get:51 http://ppa.launchpad.net/ubuntugis/ubuntugis-unstable/ubuntu trusty/main amd64 libopenjp2-7-dev amd64 2.1.0-2.1otb1~trusty1 [25.9 kB]
Get:52 http://ppa.launchpad.net/ubuntugis/ubuntugis-unstable/ubuntu trusty/main amd64 libspatialite-dev amd64 4.1.1-5ubuntu2 [948 kB]
Get:53 http://ppa.launchpad.net/ubuntugis/ubuntugis-unstable/ubuntu trusty/main amd64 libgdal-dev amd64 2.2.2+dfsg-1~trusty1 [6,337 kB]
```

and here's your build.

```
Get:23 http://us-east-1.ec2.archive.ubuntu.com/ubuntu trusty/universe amd64 libgdal1h amd64 1.10.1+dfsg-5ubuntu1 [3,699 kB]
Get:24 http://us-east-1.ec2.archive.ubuntu.com/ubuntu trusty/main amd64 libgif-dev amd64 4.1.6-11 [20.3 kB]
Get:25 http://us-east-1.ec2.archive.ubuntu.com/ubuntu trusty-updates/universe amd64 libxerces-c-dev amd64 3.1.1-5.1+deb8u4build0.14.04.1 [1,623 kB]
Get:26 http://us-east-1.ec2.archive.ubuntu.com/ubuntu trusty/main amd64 libwebp-dev amd64 0.4.0-4 [203 kB]
Get:27 http://us-east-1.ec2.archive.ubuntu.com/ubuntu trusty/universe amd64 libhdf4-alt-dev amd64 4.2r10-0ubuntu1 [338 kB]
Get:28 http://us-east-1.ec2.archive.ubuntu.com/ubuntu trusty/universe amd64 libspatialite-dev amd64 4.1.1-5ubuntu1 [938 kB]
Get:29 http://us-east-1.ec2.archive.ubuntu.com/ubuntu trusty/universe amd64 libgdal-dev amd64 1.10.1+dfsg-5ubuntu1 [4,308 kB]
Get:30 http://us-east-1.ec2.archive.ubuntu.com/ubuntu trusty/universe amd64 proj-bin amd64 4.8.0-2ubuntu2 [31.3 kB]
```

I don't think that you did anything to break it, but let me investigate further.

**Comment by jameshiebert** [comment]
OK, you should be able to merge the `hotfix/gdal-ppa` branch into here and it will fix your Travis problems.

**Pull request merged**